packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,428 @@
+      await flushPromises();
+
+      const event = new CustomEvent(
+        'quantic__registerrecommendationtemplates',
+        {
+          bubbles: true,
+          detail: {
+            content: defaultOptions.recommendation,
+            conditions: [],
+          },
+        }
+      );
+      element.shadowRoot.dispatchEvent(event);
+
+      expect(functionMocks.registerTemplates).toHaveBeenCalledTimes(1);
+    });
+
+    it('should register the fields to include', async () => {
+      const expectedFormattedFieldsToInclude = ['field1', 'field2'];
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledTimes(1);
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionMocks.dispatch).toHaveBeenCalledTimes(1);
+      expect(functionMocks.dispatch).toHaveBeenCalledWith(
+        functionMocks.registerFieldsToInclude(expectedFormattedFieldsToInclude)
+      );
+    });
+
+    describe('with custom #numberOfRecommendations option', () => {
+      it('should build the buildRecommendationList controller with the custom number of recommendation value', async () => {
+        createTestComponent({
+          ...defaultOptions,
+          numberOfRecommendations: 5,
+        });
+        await flushPromises();
+
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+          exampleEngine,
+          {
+            options: {
+              id: defaultOptions.recommendation,
+              numberOfRecommendations: Number(5),
+            },
+          }
+        );
+      });
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized successfully', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    describe('when there are no recommendations', () => {
+      it('should not display a recommendation list', async () => {
+        recommendationListState = {
+          isLoading: false,
+          recommendations: [],
+          error: null,
+          searchResponseId: null,
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const recommendationContainer = element.shadowRoot.querySelector(
+          selectors.gridContainer
+        );
+
+        expect(recommendationContainer).toBeNull();
```

Comment:
I think I would add an expect to make sure there are 0 results displayed too. That's kinda the important part :)